packages/devtools_app/lib/src/shared/table/table.dart

```diff
@@ -811,6 +806,25 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
     scrollController = widget.tableController.verticalScrollController!;
 
     pinnedScrollController = ScrollController();
+  }
+
+  @override
+  void didUpdateWidget(covariant _Table<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_data == oldWidget.tableController.tableData.value.data) return;
+    _initData();
+  }
+
+  void _initData() {
+    cancelListeners();
```

Comment:
cancelListeners doesn't need to be called in initState. Instead of including this line in the helper, add it to didUpdateWidget directly.